js/modules/k6/grpc/reflect.go

```diff
@@ -0,0 +1,41 @@
+package grpc
+
+import (
+	"crypto/tls"
+	"database/sql"
+	"net"
+	"time"
+
+	"github.com/loadimpact/k6/lib"
+	"github.com/loadimpact/k6/lib/netext"
+	"github.com/loadimpact/k6/lib/types"
+	reflectpb "google.golang.org/grpc/reflection/grpc_reflection_v1alpha"
+	"gopkg.in/guregu/null.v3"
+)
+
+func doReflect(
+	client reflectpb.ServerReflection_ServerReflectionInfoClient,
+	req *reflectpb.ServerReflectionRequest) (*reflectpb.ServerReflectionResponse, error) {
+	if err := client.Send(req); err != nil {
+		return nil, err
+	}
+	return client.Recv()
+}
+
+func reflectState() *lib.State {
+	return &lib.State{
+		TLSConfig: &tls.Config{
+			MinVersion: tls.VersionTLS10,
+			MaxVersion: tls.VersionTLS13,
+		},
+		Options: lib.Options{UserAgent: null.String{NullString: sql.NullString{
+			String: "k6",
+			Valid:  true,
+		}}, HTTPDebug: null.String{}},
+		Dialer: netext.NewDialer(
+			net.Dialer{
+				Timeout:   2 * time.Second,
+				KeepAlive: 10 * time.Second,
+			}, netext.NewResolver(net.LookupIP, 0, types.DNSfirst, types.DNSpreferIPv4)),
```

Comment:
Hhmm shouldn't all of this inherit from the global settings (`state.Options`)? The `UserAgent` for sure should be consistent with the one set for HTTP and [eventually WS :sweat_smile:](https://github.com/k6io/k6/issues/2036), and the Dialer/TLS/DNS settings also seem out of place here, since these should be configurable and not static.

Hopefully this wouldn't be a big problem and we can fix it before #883.

Reply:
Now it's in the VU context and this has been removed